tests/integrations/persisters/test_mongodb_persister.py

```diff
@@ -0,0 +1,34 @@
+import unittest
+
+from burr.core import state
+from burr.integrations.persisters.b_mongodb import MongoDBPersister
+
+
+class TestMongoDBPersister(unittest.TestCase):
+    def setUp(self):
+        self.persister = MongoDBPersister(
+            uri="mongodb://localhost:27017", db_name="testdb", collection_name="testcollection"
+        )
+
+    def tearDown(self):
+        self.persister.collection.drop()
+
+    def test_save_and_load_state(self):
+        self.persister.save("pk", "app_id", 1, "pos", state.State({"a": 1, "b": 2}), "completed")
+        data = self.persister.load("pk", "app_id", 1)
+        self.assertEqual(data["state"].get_all(), {"a": 1, "b": 2})
+
+    def test_list_app_ids(self):
+        self.persister.save("pk", "app_id1", 1, "pos1", state.State({"a": 1}), "completed")
+        self.persister.save("pk", "app_id2", 2, "pos2", state.State({"b": 2}), "completed")
+        app_ids = self.persister.list_app_ids("pk")
+        self.assertIn("app_id1", app_ids)
+        self.assertIn("app_id2", app_ids)
+
+    def test_load_nonexistent_key(self):
+        state_data = self.persister.load("pk", "nonexistent_key")
+        self.assertIsNone(state_data)
+
+
+if __name__ == "__main__":
+    unittest.main()
```

Comment:
Shoot I apologize for missing this earlier, but we use pytest and not unittest.

This shouldn't be a big change to get it to work. You'll just need to use a pytest fixture for the persister.

Reply:
i dont know weather i did it as required. help me if it is not correct

Reply:
Close - can you rename the file to test_b_mongodb.py please? that's the pytest convention. `test_NAME_OF_MODULE.py`.

Otherwise we will need to set up the test environment to install mongodb for the test to actually pass.

Why don't we try adjusting part of `python-package.yml` in .github/workflows to be something like this:

```
  test:
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        python-version: ['3.9', '3.10', '3.11', '3.12']

    steps:
      - uses: actions/checkout@v3
      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v3
        with:
          python-version: ${{ matrix.python-version }}
      - name: Install dependencies
        run: |
          python -m pip install -e ".[tests,tracking-client]"
      - name: Run tests
        run: |
          python -m pytest tests --ignore=tests/integrations/persisters
  test-persister-dbs:
    runs-on: ubuntu-latest
    services:
      mongodb:
        image: mongo:7.0
        ports:
          - 27017:27017
      postgres:
        image: postgres:15
        ports:
          - 5432:5432
      redis:
        image: redis:7
        ports:
          - 6379:6379
    strategy:
      fail-fast: false
      matrix:
        python-version: ['3.9', '3.10', '3.11', '3.12']

    steps:
      - uses: actions/checkout@v3
      - name: Set up Python ${{ matrix.python-version }}
        uses: actions/setup-python@v3
        with:
          python-version: ${{ matrix.python-version }}
      - name: Install dependencies
        run: |
          python -m pip install -e ".[tests,tracking-client]"
      - name: Run tests
        run: |
          python -m pytest tests/integrations/persisters/

```

Can you add that please?

Reply:
I did that, please let me know whether it was correct or not.